### PR TITLE
[master < T154-MAGE] Add k-means clustering docs 

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

Add docs for the `k-means` procedure on the MAGE project.

### Pull request type
- [x] New documentation page

### Checklist:
- [x] I checked all content with Grammarly
- [x] I performed a self-review of my code
- [x] I made corresponding changes to the rest of the documentation
- [x] The build passes locally
- [x] My changes generate no new warnings or errors

### CODE REVIEW

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+id: kmeans-clustering-algorithm
+title: K-Means Clustering
+sidebar_label: K-Means Clustering
+---
+
+## Description
+
+The KMeans algorithm clusters given data by trying to separate samples in `n` groups of equal variance, by minimizing a criterion known as
```
Comment (vpavicic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:9]:
```suggestion
The K-Means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing a criterion known as 
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+
+
+### Implementation
+
+[![K-Means](https://img.shields.io/badge/KMeans-Implementation-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/kmeans.py)
+
+[![K-Means](https://img.shields.io/badge/KMeans-Documentation-FCC624?style=for-the-badge&logo=python&logoColor=white)](/mage/query-modules/python/kmeans-clustering)
+
+**K-Means** algorithm is implemented within project
```
Comment (vpavicic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:27]:
```suggestion
**K-Means** algorithm is implemented within the
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+[**MAGE**](https://github.com/memgraph/mage). Be sure to check it out in the
```
Comment (vpavicic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:28]:
```suggestion
[**MAGE**](https://github.com/memgraph/mage) project. Be sure to check it out in the
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+link above.
+
+## Use cases
+
+[![Social
+networks](https://img.shields.io/badge/Social_networks-Application-8A477F?style=for-the-badge)](/use-cases/social-media.md)
+
+Biggest use case for clustering is in social networks. There communities
+can be explored, similarities found, and many more.
```
Comment (vpavicic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:37]:
```suggestion
Clustering is mostly used in social networks. Communities
can be explored, similarities found, and much more.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    }}>
+{children}
+</span>
+);
+
+[![docs-source](https://img.shields.io/badge/source-node2vec-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/kmeans.py)
+
+## Abstract
+
+The KMeans algorithm clusters given data by trying to separate samples in `n` groups of equal variance, by minimizing a criterion known as
```
Comment (vpavicic) [inline mage/query-modules/python/kmeans-clustering.md:26]: Please choose how we are going to call this algorithm as I can see there are a lot of variants (kMeans, k-means, K-means, K-Means) and stick to it :)

File: mage/query-modules/python/kmeans-clustering.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+## Procedures
+
+### `get_clusters( n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
+This procedure returns for each node to which cluster it belongs to.
+
+#### Input:
+
+- `n_clusters : int` ➡ The number of clusters to form.
+- `embedding_property : str` ➡ From which property of node will embeddings be read.
```
Comment (vpavicic) [inline mage/query-modules/python/kmeans-clustering.md:45]:
```suggestion
- `embedding_property : str` ➡ Node property used to read embeddings.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+- `init : str` ➡ Method for initialization. If `k-means++` is selected, then initial cluster centroids  are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence, and is theoretically proven to be `O(logk)`-optimal. 
+If `random` then choose `n_clusters` observations (rows) at random from data for the initial centroids.
+- `n_init : int` ➡ Number of time the k-means algorithm will be run with different centroid seeds
+- `max_iter : int` ➡ Length of one walk in walk sampling.
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`
+- `random_state : int` ➡ Use an int to make the randomness deterministic
```
[the remark this answers is not in the document]
Comment (vpavicic) [reply mage/query-modules/python/kmeans-clustering.md:52]: I used to think I was smart, then I read this: If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia.

:D :D

File: mage/query-modules/python/kmeans-clustering.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+```cypher
+ CALL kmeans.get_clusters(2, "embedding", "k-means++", 10, 10, 0.0001, "auto", 1) YIELD node, cluster_id
+ RETURN node.id as node_id, cluster_id
+   ORDER BY node_id ASC;
+```
+
+### `set_clusters( n_clusters, embedding_property, cluster_property, init, n_init, max_iter, tol, algorithm, random_state)`
+Procedure sets for each node to which cluster it belongs to by writing cluster id  to `cluster_property`.
+
+#### Input:
```
Comment (vpavicic) [inline mage/query-modules/python/kmeans-clustering.md:70]: Plase check the changes I made in a similar block above, and apply the same changes here

File: mage/query-modules/python/kmeans-clustering.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+If `random` then choose `n_clusters` observations (rows) at random from data for the initial centroids.
+- `n_init : int` ➡ Number of time the k-means algorithm will be run with different centroid seeds
+- `max_iter : int` ➡ Length of one walk in walk sampling.
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`
+- `random_state : int` ➡ Use an int to make the randomness deterministic
+
+#### Output:
+
+- `node: mgp.Vertex` ➡ Vertex for which we got cluster_id
```
Comment (vpavicic) [inline mage/query-modules/python/kmeans-clustering.md:85]: Please check that each sentence ends with a dot (.)

**Review by vpavicic — CHANGES_REQUESTED** [review]
Some grammar issues and consistencies

File: mage/query-modules/python/kmeans-clustering.md (hunk 5 above)
Comment (antoniofilipovic) [reply mage/query-modules/python/kmeans-clustering.md:26]: I will stick to` k-means`

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+sidebar_label: K-means Clustering
+---
+
+## Description
+
+The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing a criterion known as 
+within-the-cluster sum-of-squares. 
+Algorithm requires the number of clusters to be specified beforehand. This means it is not possible for algorithm to determine
+what is the best number of clusters to use, but it is up to user to find it. It scales well to large numbers of samples and has been used 
+across a large range of application areas in many different fields such as recommendation engine, fraud detection and so on.
```
Comment (antepusic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:13]: Correcting grammar + redundancies
```suggestion
The algorithm requires the number of clusters to be specified beforehand. In other words, the algorithm doesn’t compute the optimal number of clusters to use, but leaves that task to the user. It scales well to large numbers of samples, and it has been applied in a wide range of fields such as recommendation engines, fraud detection and market research.
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
+---
+id: k-means-clustering-algorithm
+title: K-means Clustering
+sidebar_label: K-means Clustering
+---
+
+## Description
+
+The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing a criterion known as
```
Comment (antepusic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:9]:
```suggestion
The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing the criterion known as 
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing a criterion known as 
+within-the-cluster sum-of-squares. 
+Algorithm requires the number of clusters to be specified beforehand. This means it is not possible for algorithm to determine
+what is the best number of clusters to use, but it is up to user to find it. It scales well to large numbers of samples and has been used 
+across a large range of application areas in many different fields such as recommendation engine, fraud detection and so on.
+
+
+![k-means clustering](../../data/algorithms/machine-learning-graph-analytics/kmeans-clustering.png)
+> K-means clustering run on top of Node2Vec algorithm. Numbers represent cluster.
```
Comment (antepusic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:17]:
```suggestion
> K-means clustering run on node2vec output; clusters are numbered.
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+### Implementation
+
+[![k-means](https://img.shields.io/badge/KMeans-Implementation-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/kmeans.py)
+
+[![k-means](https://img.shields.io/badge/KMeans-Documentation-FCC624?style=for-the-badge&logo=python&logoColor=white)](/mage/query-modules/python/kmeans-clustering)
+
+**K-Means** algorithm is implemented within the
+[**MAGE**](https://github.com/memgraph/mage) project. Be sure to check it out in the
+link above.
```
Comment (antepusic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:29]:
```suggestion
The **K-Means** algorithm is implemented within the
[**MAGE**](https://github.com/memgraph/mage) project. Be sure to check it out in the
link above.
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+## Use cases
+
+[![Social
+networks](https://img.shields.io/badge/Social_networks-Application-8A477F?style=for-the-badge)](/use-cases/social-media.md)
+
+Clustering is mostly used in social networks. Communities
+can be explored, similarities found, and much more.
```
Comment (antepusic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:37]:
```suggestion
Clustering is heavily used in social network analysis to discover user communities, find similarities between them and much more.
```

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md (hunk 12 above)
Comment (antepusic) [reply mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:17]: BTW, is the node2vec part important here? I’ve looked at the above image and don’t see how it relates to node2vec

File: mage/query-modules/python/kmeans-clustering.md [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+| ------------------- | -------------------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**module**</Highlight>              |
+| **Implementation**  | <Highlight color="#FB6E00">**Python**</Highlight>              |
+| **Graph direction** | <Highlight color="#FB6E00">**directed/undirected**</Highlight> |
+| **Edge weights**    | <Highlight color="#FB6E00">**weighted/unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>          |
+
+## Procedures
+
+### `get_clusters( n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:39]: Spacing
```suggestion
### `get_clusters(n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+    }}>
+{children}
+</span>
+);
+
+[![docs-source](https://img.shields.io/badge/source-kmeans-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/kmeans.py)
+
+## Abstract
+
+The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance, by minimizing a criterion known as
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:26]:
```suggestion
The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing the criterion known as 
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+{children}
+</span>
+);
+
+[![docs-source](https://img.shields.io/badge/source-kmeans-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/kmeans.py)
+
+## Abstract
+
+The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance, by minimizing a criterion known as 
+within-the-cluster sum-of-squares. To understand more about it, jump to the [algorithm](../../algorithms/machine-learning-graph-analytics/k-means-clustering-algorithm) page.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:27]: Rather don’t tell the readers they don’t understand it
```suggestion
within-the-cluster sum-of-squares. To learn more about it, jump to the [algorithm](../../algorithms/machine-learning-graph-analytics/k-means-clustering-algorithm) page.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+| **Module type**     | <Highlight color="#FB6E00">**module**</Highlight>              |
+| **Implementation**  | <Highlight color="#FB6E00">**Python**</Highlight>              |
+| **Graph direction** | <Highlight color="#FB6E00">**directed/undirected**</Highlight> |
+| **Edge weights**    | <Highlight color="#FB6E00">**weighted/unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>          |
+
+## Procedures
+
+### `get_clusters( n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
+This procedure returns for each node to which cluster it belongs to.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:40]: Gramar
```suggestion
For each node, this procedure returns what cluster it belongs to.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+
+- `n_clusters : int` ➡ The number of clusters to form.
+- `embedding_property : str` ➡ Node property where embeddings are stored.
+- `init : str` ➡ Initialization method. If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal. 
+If `random` then `n_clusters` observations (rows) are chosen for the initial centroids randomly from the data.
+- `n_init : int` ➡ Number of times the k-means algorithm will be run with different centroid seeds.
+- `max_iter : int` ➡ Length of one walk in walk sampling.
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`.
+- `random_state : int` ➡ Use an int to make the randomness deterministic.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:52]:
```suggestion
- `random_state : int` ➡ Random seed for the algorithm.
```
By the way, does this parameter have a default value?

File: mage/query-modules/python/kmeans-clustering.md [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+#### Input:
+
+- `n_clusters : int` ➡ The number of clusters to form.
+- `embedding_property : str` ➡ Node property where embeddings are stored.
+- `init : str` ➡ Initialization method. If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal. 
+If `random` then `n_clusters` observations (rows) are chosen for the initial centroids randomly from the data.
+- `n_init : int` ➡ Number of times the k-means algorithm will be run with different centroid seeds.
+- `max_iter : int` ➡ Length of one walk in walk sampling.
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:51]: Where are the options described? Maybe we could use a link here.

File: mage/query-modules/python/kmeans-clustering.md [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>          |
+
+## Procedures
+
+### `get_clusters( n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
+This procedure returns for each node to which cluster it belongs to.
+
+#### Input:
+
+- `n_clusters : int` ➡ The number of clusters to form.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:44]:
```suggestion
- `n_clusters : int` ➡ Number of clusters to be formed.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 22]
```diff
@@ -0,0 +1,171 @@
...
+- `embedding_property : str` ➡ Node property where embeddings are stored.
+- `init : str` ➡ Initialization method. If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal. 
+If `random` then `n_clusters` observations (rows) are chosen for the initial centroids randomly from the data.
+- `n_init : int` ➡ Number of times the k-means algorithm will be run with different centroid seeds.
+- `max_iter : int` ➡ Length of one walk in walk sampling.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:49]:
```suggestion
- `max_iter : int` ➡ Length of sampling walks.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 23]
```diff
@@ -0,0 +1,171 @@
...
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:50]: Clarify
```suggestion
- `tol : float` ➡ Relative tolerance of the Frobenius norm of the difference of cluster centers across consecutive iterations. Used in determining convergence.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 24]
```diff
@@ -0,0 +1,171 @@
...
+## Procedures
+
+### `get_clusters( n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
+This procedure returns for each node to which cluster it belongs to.
+
+#### Input:
+
+- `n_clusters : int` ➡ The number of clusters to form.
+- `embedding_property : str` ➡ Node property where embeddings are stored.
+- `init : str` ➡ Initialization method. If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:46]:
```suggestion
- `init : str` ➡ Initialization method. If `k-means++` is selected, initial cluster centroids are sampled per an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal. 
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 25]
```diff
@@ -0,0 +1,171 @@
...
+
+### `get_clusters( n_clusters, embedding_property, init, n_init, max_iter, tol, algorithm, random_state)`
+This procedure returns for each node to which cluster it belongs to.
+
+#### Input:
+
+- `n_clusters : int` ➡ The number of clusters to form.
+- `embedding_property : str` ➡ Node property where embeddings are stored.
+- `init : str` ➡ Initialization method. If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal. 
+If `random` then `n_clusters` observations (rows) are chosen for the initial centroids randomly from the data.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:47]: What do you mean by "rows"?
```suggestion
If `random`, `n_clusters` observations (rows) are randomly chosen for the initial centroids.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 26]
```diff
@@ -0,0 +1,171 @@
...
+- `n_init : int` ➡ Number of times the k-means algorithm will be run with different centroid seeds.
+- `max_iter : int` ➡ Length of one walk in walk sampling.
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`.
+- `random_state : int` ➡ Use an int to make the randomness deterministic.
+
+#### Output:
+
+- `node: mgp.Vertex` ➡ Vertex for which we got cluster_id.
+- `cluster_id: mgp.Number` ➡ Corresponding cluster id of vertex.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:57]:
```suggestion
- `node: mgp.Vertex` ➡ Graph node.
- `cluster_id: mgp.Number` ➡ Cluster ID of the above node.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 27]
```diff
@@ -0,0 +1,171 @@
...
+- `n_clusters : int` ➡ The number of clusters to form.
+- `embedding_property : str` ➡ Node property where embeddings are stored.
+- `cluster_property: str` ➡ Node property where cluster_id will be stored.
+- `init : str` ➡ Initialization method. If `k-means++` is selected, the initial cluster centroids are selected using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence and is theoretically proven to be `O(logk)`-optimal. 
+If `random` then `n_clusters` observations (rows) are chosen for the initial centroids randomly from the data.
+- `n_init : int` ➡ Number of times the k-means algorithm will be run with different centroid seeds.
+- `max_iter : int` ➡ Length of one walk in walk sampling.
+- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`.
+- `random_state : int` ➡ Use an int to make the randomness deterministic.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:81]: The same suggestions apply as with `get_clusters`

File: mage/query-modules/python/kmeans-clustering.md [hunk 28]
```diff
@@ -0,0 +1,171 @@
...
+
+#### Output:
+
+- `node: mgp.Vertex` ➡ Vertex for which we got cluster_id.
+- `cluster_id: mgp.Number` ➡ Corresponding cluster id of vertex.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:86]: The same suggestions apply as with `get_clusters`

**Review by antepusic — CHANGES_REQUESTED** [review]
I’ve left suggestions to make wording clearer (configuring _k_-means can get fairly technical). Also, make sure you add a short description to the README.

File: mage/query-modules/python/kmeans-clustering.md (hunk 25 above)
Comment (antoniofilipovic) [reply mage/query-modules/python/kmeans-clustering.md]: Yes, if I am not wrong, observations are basically data points

Comment (antoniofilipovic) [reply mage/query-modules/python/kmeans-clustering.md]: Rows meaning one data point in 2d table where row is each entry

File: mage/query-modules/python/kmeans-clustering.md (hunk 19 above)
Comment (antoniofilipovic) [reply mage/query-modules/python/kmeans-clustering.md]: I think it does! I will update it

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md (hunk 12 above)
Comment (antoniofilipovic) [reply mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md]: It is not important, I will remove it

File: mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md [hunk 29]
```diff
@@ -6,25 +6,21 @@ sidebar_label: K-means Clustering
...
 
 ### Implementation
 
 [![k-means](https://img.shields.io/badge/KMeans-Implementation-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/kmeans.py)
 
-[![k-means](https://img.shields.io/badge/KMeans-Documentation-FCC624?style=for-the-badge&logo=python&logoColor=white)](/mage/query-modules/python/kmeans-clustering)
+[![k-means](https://img.shields.io/badge/KMeans-Documentation-FCC624?style=for-the-badge&logo=python&logoColor=white)](/mage/query-modules/python/kmeans)
 
-**K-Means** algorithm is implemented within the
+**K-means** algorithm is implemented within the
```
Comment (antepusic) [inline mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md:23]:
```suggestion
The **K-means** algorithm is implemented within the
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 30]
```diff
@@ -69,21 +69,21 @@ Procedure sets for each node to which cluster it belongs to by writing cluster i
 
 #### Input:
 
-- `n_clusters : int` ➡ The number of clusters to form.
+- `n_clusters : int` ➡ Number of clusters to be formed.
 - `embedding_property : str` ➡ Node property where embeddings are stored.
 - `cluster_property: str` ➡ Node property where cluster_id will be stored.
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:74]:
```suggestion
- `cluster_property: str` ➡ Node property where `cluster_id` will be stored.
```

File: mage/query-modules/python/kmeans-clustering.md [hunk 31]
```diff
@@ -36,25 +36,25 @@ within-the-cluster sum-of-squares. To understand more about it, jump to the [alg
...
+If `random`, `n_clusters` observations (rows) are randomly chosen for the initial centroids.
 - `n_init : int` ➡ Number of times the k-means algorithm will be run with different centroid seeds.
-- `max_iter : int` ➡ Length of one walk in walk sampling.
-- `tol : float` ➡ Relative tolerance with regards to Frobenius norm of the difference in the cluster centers of two consecutive iterations to declare convergence.
-- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`.
-- `random_state : int` ➡ Use an int to make the randomness deterministic.
+- `max_iter : int` ➡ Length of sampling walks.
+- `tol : float` ➡ Relative tolerance of the Frobenius norm of the difference of cluster centers across consecutive iterations. Used in determining convergence.
+- `algorithm : str` ➡ Options are `lloyd`, `elkan`, `auto`, `full`. Description [here](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html#:~:text=algorithm%7B%E2%80%9Clloyd%E2%80%9D%2C%20%E2%80%9Celkan%E2%80%9D%2C%20%E2%80%9Cauto%E2%80%9D%2C%20%E2%80%9Cfull%E2%80%9D%7D%2C%20default%3D%E2%80%9Dlloyd%E2%80%9D).
+- `random_state : int` ➡ Random seed for the algorithm (default:1998).
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:52]: Please add the default values from https://github.com/memgraph/mage/blob/main/python/kmeans.py.
Also, make the default value format consistent with existing query modules (e.g. [katz_centrality_online](https://memgraph.com/docs/mage/query-modules/cpp/katz-centrality-online#setalpha-epsilon)) - the values are easy to miss if they’re to the right of the description.

File: mage/query-modules/python/kmeans-clustering.md [hunk 32]
```diff
@@ -69,21 +69,21 @@ Procedure sets for each node to which cluster it belongs to by writing cluster i
...
```
Comment (antepusic) [inline mage/query-modules/python/kmeans-clustering.md:81]: Same thing as with `get_clusters`

**Review by antepusic — CHANGES_REQUESTED** [review]
Almost done! I caught a few bits that need changes.

File: mage/query-modules/python/kmeans-clustering.md (hunk 31 above)
Comment (antoniofilipovic) [reply mage/query-modules/python/kmeans-clustering.md]: Done!

File: mage/query-modules/python/kmeans-clustering.md (hunk 32 above)
Comment (antoniofilipovic) [reply mage/query-modules/python/kmeans-clustering.md]: Done!

**Review by antepusic — APPROVED** [review]

**Pull request merged**